proto/api/v1/workflow.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package api.v1;
+
+import "google/rpc/status.proto";
+
+message ExecuteWorkflowRequest {
+  // URL of the repo the workflow is running for
+  // Ex. "https://github.com/some-user/acme"
+  string repo_url = 1;
+  // Branch the workflow should run on
+  // Ex. "cool-feature"
+  string branch = 2;
```

Comment:
1. I would suggest renaming this to a `ref` as it should behave the same for git branch as well as git tag.

2. I wonder if we should use commit instead of ref. Reason being that there could be a small window between the time a Workflow was requested and the actual repo being fetch where user could force push and update a branch. In such case, it might be a lot less confusing to have separate builds. Such branch could be rapidly updated by user, or is rapidly updated main branch in a monorepo with thousands of user.

Reply:
Technically `ref` is a pointer to `commit_sha`, so `commit_sha` alone should suffice.

However, our Web UI could really use the additional data from `ref` to help group the invocations under the same branch/tag together. Therefore, I prefer option (b). 

I don't see any scenarios where you would _NOT_ want to supply a branch/tag name to BuildBuddy.
In fact, I wonder if we should be adding an additional K/V field to allow additional Build Metadata for each Workflow invocation 🤔

Reply:
Talked offline. For now, will just have a `ref` field and not a `commit_sha` field and translate the ref to a commit sha in the backend

We're expecting that for the majority of use cases of this manual API, people will want the top commit. This will save them the tedium of having to grab it themselves. We can easily add an optional commit field if it's requested